### PR TITLE
Introducing tear gas

### PR DESCRIPTION
**Pull request by Jawshy**
Introducing tear gas effects as suggested in #366. fn_tearGas authored by larrow and retrieved from [Teargas / Toxic Gas Script Help](https://forums.bistudio.com/topic/180844-teargas-toxic-gas-script-help/?p=2850713). 

###### Config_Clothing.hpp

- Added: 
  - H_CrewHelmetHeli_B / Heli Crew Helmet [NATO] to serve as a gas mask. Available to police rank 3 and above. 

###### Config_SpyGlass.hpp

- Updated: 
  - allowedVariables to include `{"fnc_insmoke","CODE"}`. 

###### Config_Weapons.hpp

- Added: 
  - SmokeShellYellow / Smoke Grenade (Yellow) to serve as a tear gas grenade. Available to police rank 3 and above. 

###### core\init.sqf 

- Added: 
  - `[] execVM "core\items\fn_tearGas.sqf";`

###### core\items\fn_tearGas.sqf

- Added: 
  - fn_tearGas script by larrow with minor modifications. 

###### addMagazineCargo.txt

- Added: 
  - `!=SmokeShellYellow`

###### createVehicle.txt

- Added
  - `!=SmokeShellYellow`

Note: fn_tearGas includes `player setVariable [ "inSmoke", true ];` however BEFilters did not log `inSmoke` during testing and thus it was not added to setVariable.txt.

### CODE REVIEW

File: Altis_Life.Altis/core/items/fn_tearGas.sqf [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+gasMasks = ["H_CrewHelmetHeli_B"];
+
+//Are we near a smoke shell.
+fnc_smokeNear = {
+
+	//Are we not wearing a gas mask.
+	if ( !( headgear player in gasMasks ) ) then {
+
+		//Get nearest smoke shell.
+		_smokeShell = nearestObject [ getPosATL player, "SmokeShellYellow" ];
```
Comment (sargken) [inline Altis_Life.Altis/core/items/fn_tearGas.sqf]: Shouldnt it be white?

**Comment by trinitynet** [comment]
i don't like the idea of having more While {true} 's kicking about would it not be better to have an event handler fire when someone throws a gas nade? also why it it not structured like the rest of the life system? IE: life_fnc_???

File: Altis_Life.Altis/core/items/fn_tearGas.sqf (hunk 1 above)
Comment (fEcAlMaTteR) [inline Altis_Life.Altis/core/items/fn_tearGas.sqf]: If it were white a player wouldn't be able to tell the difference from normal smoke.

Comment (sargken) [inline Altis_Life.Altis/core/items/fn_tearGas.sqf]: True

**Comment by Kimax89** [comment]
I will be testing this on my server soon.

**Comment by trinitynet** [comment]
i can confirm it works

**Comment by fEcAlMaTteR** [comment]
I would suggest a configuration to allow it to not init.  For people who wished not to use it an extra event on each frame is un-needed tax.

**Comment by Kimax89** [comment]
Confirmed working from me.

**Comment by trinitynet** [comment]
i can do this if noone is there to take it on?

**Comment by trinitynet** [comment]
double tapping escape bypasses the gas

**Comment by Jawshy** [comment]
> i can do this if noone is there to take it on?

@trinitynet, go for it.

**Comment by trinitynet** [comment]
see #414

**Comment by Kimax89** [comment]
@trinitynet Your pull is just pure mess, but i'm looking forward to see a clean pull for this one.

@Jawshy Does this affect all the yellow smoke? even the ones from the granade launcher? or do i have to edit it?

**Comment by Jawshy** [comment]
No. `SmokeShellYellow` is for [Smoke Grenade (Yellow)](https://community.bistudio.com/wiki/Arma_3_CfgMagazines#SmokeShellYellow). [Smoke Round (Yellow)](https://community.bistudio.com/wiki/Arma_3_CfgMagazines#1Rnd_SmokeYellow_Grenade_shell) and [3Rnd 3GL Smoke Rounds (Yellow)](https://community.bistudio.com/wiki/Arma_3_CfgMagazines#3Rnd_SmokeYellow_Grenade_shell) are there own classes.

**Comment by danielstuart14** [comment]
should be added as a trigger

**Pull request closed without merging**